src/main/java/com/google/sps/servlets/EditBusinessDataServlet.java

```diff
@@ -0,0 +1,248 @@
+    String address = request.getParameter(BUSINESS_ADDRESS);
+    String addressLatStr = request.getParameter(BUSINESS_ADDRESS_LAT);
+    String addressLngStr = request.getParameter(BUSINESS_ADDRESS_LNG);
+    float addressLat;
+    if (addressLatStr.isEmpty()) {
+        addressLat = 404;
+    } else {
+        addressLat = Float.parseFloat(addressLatStr);
+    }
+    float addressLng;
+    if (addressLngStr.isEmpty()) {
+        addressLng = 404;
+    } else {
+        addressLng = Float.parseFloat(addressLngStr);
+    }
+    String contactDetails = request.getParameter(BUSINESS_CONTACT_INFO);
+    String orderDetails = request.getParameter(BUSINESS_ORDER_INFO);
+    String minPriceStr = request.getParameter(BUSINESS_MIN_PRICE);
+    String maxPriceStr = request.getParameter(BUSINESS_MAX_PRICE);
+    float minPrice;
+      if (minPriceStr.isEmpty()) {
+          minPrice = 404;
+      } else {
+          minPrice = Float.parseFloat(minPriceStr);
+      }
+      float maxPrice;
+      if (maxPriceStr.isEmpty()) {
+          maxPrice = 404;
+      } else {
+          maxPrice = Float.parseFloat(maxPriceStr);
+      }
+
+    String businessLink = request.getParameter(BUSINESS_LINK);
+    String menuLink = request.getParameter(BUSINESS_MENU_LINK);
+    String logoUrl = getUploadedLogoUrlFromBlobstore(request, BUSINESS_LOGO);
+    List<String> picturesUrls = getUploadedPicturesUrlsFromBlobstore(request, BUSINESS_PICTURES);
+    // can't add reviews and rating when creating a new business
+    float rating = 404;
+    List<String> reviews = new ArrayList<String>();
+
+    Entity businessEntity = getBusinessEntity(datastore, businessKey);
+    
+    businessEntity.setProperty(BUSINESS_NAME, name);
+    businessEntity.setProperty(BUSINESS_DESC, desc);
+    businessEntity.setProperty(BUSINESS_CATEGORIES, gson.toJson(categories));
+    businessEntity.setProperty(BUSINESS_ADDRESS, address);
+    businessEntity.setProperty(BUSINESS_ADDRESS_LAT, addressLat);
+    businessEntity.setProperty(BUSINESS_ADDRESS_LNG, addressLng);
+    businessEntity.setProperty(BUSINESS_CONTACT_INFO, contactDetails);
+    businessEntity.setProperty(BUSINESS_ORDER_INFO, orderDetails);
+    businessEntity.setProperty(BUSINESS_LINK, businessLink);
+    businessEntity.setProperty(BUSINESS_MENU_LINK, menuLink);
+    businessEntity.setProperty(BUSINESS_MIN_PRICE, minPrice);
+    businessEntity.setProperty(BUSINESS_MAX_PRICE, maxPrice);
+    businessEntity.setProperty(BUSINESS_LOGO, logoUrl);
+    businessEntity.setProperty(BUSINESS_PICTURES, gson.toJson(picturesUrls));
+    businessEntity.setProperty(BUSINESS_RATING, rating);
+    businessEntity.setProperty(BUSINESS_REVIEWS, reviews);
+
+    storeBusinessEntity(datastore, businessEntity);
+
+    // Redirect back to the product page page.
+    response.sendRedirect("/index.html");
+    } catch (IOException err) {
+      System.out.println(err);
+    } catch (EntityNotFoundException err) {
+      System.out.println(err);
```

Comment:
Again, should we redirect to a "Business not found" page? I think for MVP we can keep it as a console log, but maybe add a TODO or a kanban task to improve this in the future

Reply:
okay added